action.yml

```
@@ -234,6 +239,19 @@ runs:
           fi
         fi
 
+    - uses: actions/github-script@v7
+      id: requirements-hash
+      with:
+        script: return require('fs').createReadStream(require('path').join(process.env.GITHUB_ACTION_PATH, 'requirements.txt')).pipe(require('crypto').createHash('sha1').setEncoding('hex'), 'finish').digest('hex')
+        result-encoding: string
+
+    - uses: actions/cache@v4
+      if: ${{ inputs.cache == 'true' }}
+      with:
+        key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ steps.requirements-hash.outputs.result }}
```

This key needs to be more specific to cpp-linter-action.


```suggestion
        key: cpp-linter-action-venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ steps.requirements-hash.outputs.result }}
```